code/modules/projectiles/guns/flamer.dm

```diff
@@ -491,9 +491,15 @@
 	if(istype(M))
 		M.flamer_fire_crossed(burnlevel, firelevel)
 
+/obj/flamer_fire/CanAllowThrough(atom/movable/mover, turf/target)
+	. = ..()
+	if(isxenohivemind(mover))
+		return FALSE
 
```

Comment:
why is this not in hivemind.dm

Reply:
Because this is flamer_fire's proc.

If it belongs in Hivemind.dm, then what was the problem with it being in Wraith.dm when Wraith used it?

I don't mind moving it, but it'd be nice to have this shit laid out straight instead of me having to guess at things and then being made to change it later.